h5p-bildetema/src/components/Button/Button.tsx

```diff
@@ -1,31 +1,62 @@
-import React from "react";
+/* eslint-disable react/jsx-props-no-spreading */
+import * as React from "react";
+import { Slot } from "@radix-ui/react-slot";
 import styles from "./Button.module.scss";
 
-type variants = "primary" | "secondary" | "underline" | "filter";
+const buttonVariants = {
+  variants: {
+    variant: {
+      default: `${styles.button} ${styles.default}`,
+      destructive: `${styles.button}`,
+      outline: `${styles.button}`,
+      secondary: `${styles.button} ${styles.secondary}`,
+      ghost: `${styles.button}`,
+      underline: `${styles.button} ${styles.underline}`,
+      filter: `${styles.button} ${styles.filter}`,
+    },
+    size: {
+      default: "",
+      sm: "",
+      lg: "",
+      icon: "",
+    },
+  },
+  defaultVariants: {
+    variant: "default",
+    size: "default",
+  },
+};
 
-export interface ButtonProps extends React.ComponentProps<"button"> {
-  children: React.ReactNode;
-  variant: variants;
+export interface ButtonProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  asChild?: boolean;
```

Comment:
Do we need this? If we don't see any use for it in the near future, it will be better to simplify the code

Reply:
removed asChild since its not used.